### DISCUSSION TITLE
Stopping/Shutting down playback for KSVideoPlayerView

### DISCUSSION THREAD

**Discussion by mazz** [kingslay/KSPlayer#662] [General]
How do I shutdown and fully reset the KSVideoPlayerView? If I add this code below in the `onPlayerDisappear` callback, the next time I open the player view it resumes from the point where it disappeared.

```
                layer.player.pause()
                layer.resetPlayer()
                layer.player.shutdown()
```

```swift
extension KSVideoPlayerView {

    init(model: LivestreamModel) {
        let sourceUrl = model.sourceUrl
        let options = MEOptions()
        #if DEBUG
        let url = URL(string: sourceUrl)!

        ...

        self.init(url: url, options: options, title: model.name) { layer in
            if let layer {

                layer.player.pause()
                layer.resetPlayer()
                layer.player.shutdown()
            }
        }

    }
```

**Comment by kingslay** [comment]
你可以不用设置onPlayerDisappear 。我自动就会清空。你看下是不是options.startPlayTime  这个值有被设置了。

**Reply by mazz** [reply]
I actually call `options.startPlayTime = 0`

but it seems to have no effect.

I put the KSVideoPlayerView inside an LNPopupUI swiftui view. Please see the attached video(you can hear the audio continue to play, which shows that the video is not disposed of)

https://github.com/kingslay/KSPlayer/assets/808934/4962c906-eade-46b2-8224-d5ddf5bf892c



```
        options.startPlayTime = 0
        ...
        self.init(url: url, options: options, title: model.name) { layer in
            if let layer {

                layer.player.pause()
                layer.resetPlayer()
                layer.player.shutdown()
            }
        }
```

**Comment by kingslay** [comment]
你看下 这个方法有被调用吗？没有的话，那就是你的写的swiftUI 有问题，导致view没有被释放。
  public static func dismantleUIView(_: UIViewType, coordinator: Coordinator) {

**Reply by mazz** [reply]
Thanks for the response. The issue was with LNPopupUI. This is the discussion and fix:

https://github.com/LeoNatan/LNPopupUI/discussions/28

After applying LNPopupUI 1.6.3, the video player is cleaned-up and stops, using code below also:
```
        self.init(url: url, options: options, title: model.name) { layer in
            if let layer {
               // empty now
            }
        }
```
